src/read_preference.ts

```diff
@@ -31,7 +33,9 @@ export interface ReadPreferenceOptions {
 }
 
 /** @public */
-export interface ReadPreferenceLikeOptions {
+export interface ReadPreferenceLikeOptions extends ReadPreferenceOptions {
+  readPreferenceTags?: TagSet[];
+  session?: ClientSession;
```

Comment:
this looks wrong, session should have nothing to do with a `ReadPreferenceLikeOptions`

Reply:
Indeed I added a new type `ReadPreferenceFromOptions` that represents this better, its public for now but I believe it will be made internal in the annotations work.